dependabot/update_lag_status.py

```diff
@@ -0,0 +1,290 @@
+                ci_status_link = "https://github.com/ballerina-platform/"+module[MODULE_NAME]+"/pulls"
+                ci_status = "[![CI status](https://img.shields.io/github/status/contexts/pulls/ballerina-platform/" + module[MODULE_NAME] + "/" + str(pr_number) + ")](" + ci_status_link + ")"
+            pending_pr = "[" + pr_id + "](" + pending_pr_link + ")"
+            
+            level = ""
+            if(idx==0):
+                level = str(current_level)
+   
+            table_row = "| " + level + " | [" + name + "](https://github.com/ballerina-platform/"+module[MODULE_NAME]+") | " + lag_button + " | " + pending_pr + " | " + ci_status + " |"
+            updated_readme += table_row + "\n"
+    return updated_readme, updated_modules
+
+
+def make_pie(val):
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    colors = [(60,121,189), (212, 241, 249)]
+    sizes = [val,100-val]
+    text = str(val) + "%"
+
+    col = [[i/255. for i in c] for c in colors]
+
+    fig, ax = plt.subplots()
+    ax.axis('equal')
+    width = 0.35
+    kwargs = dict(colors=col, startangle=180)
+    outside, _ = ax.pie(sizes, radius=1, pctdistance=1-width/2,**kwargs)
+    plt.setp( outside, width=width, edgecolor='white')
+
+    kwargs = dict(size=20, fontweight='bold', va='center')
+    ax.text(0, 0, text, ha='center', **kwargs)
+    plt.savefig('repo_status_graph.png')
+
+
+def return_updated_readme(readme):
+    updated_readme = ""
+    global all_modules
+
+    all_modules = get_module_list()
+
+    module_details_list = all_modules["modules"]
+
+    updated_readme += "# Ballerina repositories update status" + "\n"
+    declaration  = "ballerina-lang repository version **" +ballerina_lang_version + "** has updates as follows."
+    updated_readme += "| <img src=\"foo.png\" width=\"625\" title=\"Repositories updated\"/> | " + declaration + " |"+"\n"
+    updated_readme += "|:---:|:---|" +"\n"
+    updated_readme += "## Modules and Extensions packed in distribution" + "\n"
+    updated_readme += "| Level | Modules | Lag Status | Pending PR | Pending PRs CI Status |" + "\n"
+    updated_readme += "|:---:|:---:|:---:|:---:|:---:|" + "\n"
+
+    updated_readme, updated_modules_number = update_modules(updated_readme, module_details_list)
+    
+    updated_readme += "## Modules released to Central" + "\n"
+
+    updated_readme += "| Level | Modules | Lag Status | Pending PR | Pending PRs CI Status |" + "\n"
+    updated_readme += "|:---:|:---:|:---:|:---:|:---:|" + "\n"
+
+    central_modules = all_modules["central_modules"]
+
+    updated_readme, updated_modules_number_central = update_modules(updated_readme, central_modules)
+    updated_modules_number += updated_modules_number_central
+    repositories_updated = round((updated_modules_number/(len(module_details_list)+len(central_modules)))*100)
+    make_pie(repositories_updated)
+
+    return updated_readme
+
+
+def commit_changes(repo, updated_file, graph_image):
```

Comment:
We can't commit to the ballerina-release master branch as it is protected, we have to commit to a separate branch and create a pull request to master. This workflow can be seen in https://github.com/ballerina-platform/ballerina-release/blob/master/dependabot/update_version_data.py#L242

Reply:
b9fa8f57f883da210a8796cd7433486f885e127c